sphinx/util/inspect.py

```diff
@@ -241,14 +239,58 @@ def isclassmethod(
     return False
 
 
+def is_classmethod_descriptor(
+    obj: Any, cls: Any = None, name: str | None = None
+) -> TypeIs[types.ClassMethodDescriptorType]:
+    """Check if the object is a class method descriptor type.
+
+    This is typically useful to check if a built-in method is a class method.
+    """
+    if isinstance(obj, types.ClassMethodDescriptorType):
+        return True
+    if cls and name:
+        # trace __mro__ if the method is defined in parent class
+        sentinel = object()
+        for basecls in getmro(cls):
+            meth = basecls.__dict__.get(name, sentinel)
+            if meth is not sentinel:
+                return is_classmethod_descriptor(obj)
```

Comment:
Why recurse here? Is there a test case that requires recursion instead of checking `isinstance`?

Also should this be `meth` instead of `obj`?

Reply:
> Also should this be meth instead of obj?

Yes, that's a typo because I wrongly cc something.

> Why recurse here? Is there a test case that requires recursion instead of checking isinstance?

Not for this function so no need actually :')